core/src/main/java/org/opensearch/sql/executor/QueryService.java

```diff
@@ -73,15 +96,7 @@ public void execute(
       QueryType queryType,
       ResponseListener<ExecutionEngine.QueryResponse> listener) {
     try {
-      boolean calciteEnabled = false;
-      if (settings != null) {
-        calciteEnabled = settings.getSettingValue(Settings.Key.CALCITE_ENGINE_ENABLED);
-      }
-      // TODO https://github.com/opensearch-project/sql/issues/3457
-      // Calcite is not available for SQL query now. Maybe release in 3.1.0?
-      if (!calciteEnabled || relNodeVisitor == null || queryType == QueryType.SQL) {
-        executePlan(analyze(plan, queryType), PlanContext.emptyPlanContext(), listener);
-      } else {
+      if (shouldUseCalcite(queryType)) {
```

Comment:
As we add more and more conditional checks, shall we consider adding a QueryService for Calcite?

Reply:
Sure

Reply:
Separate a new `CalciteQueryService` simplify the code but lead to another problem: We cannot enable/disable Calcite on the fly since the QueryService instance created in `OpenSearchPluginModule` once.